docs/builders/vsphere-clone.mdx

```diff
@@ -45,6 +45,59 @@ necessary for this build to succeed and can be found further down the page.
 
 @include 'builder/vsphere/clone/CloneConfig-not-required.mdx'
 
+**Examples of usage:**
+
+Backwards compatible reconfiguring first network adapter only:
+
+<Tabs>
+<Tab heading="JSON">
+```json
+"network": "new_network",
+"mac_address": "..."
+```
+</Tab>
+<Tab heading="HCL2">
+```hcl
+network = new_network,
+mac_address = '...'
+```
+</Tab>
+</Tabs>
+
+Reconfiguring all network adapters:
+
+<Tabs>
+<Tab heading="JSON">
+```json
+"network": "new_network_for_network_adapter_0, new_network_for_network_adapter_1, new_network_for_network_adapter_2",
+"mac_address": "<mac_address_for_network_adapter_0>, <mac_address_for_network_adapter_1>, <mac_address_for_network_adapter_2>"
+```
+</Tab>
+<Tab heading="HCL2">
+```hcl
+network = 'new_network_for_network_adapter_0, new_network_for_network_adapter_1, new_network_for_network_adapter_2'
+mac_address = '<mac_address_for_network_adapter_0>, <mac_address_for_network_adapter_1>, <mac_address_for_network_adapter_2>'
```

Comment:
> First I though about introducing network_adapters from vsphere-iso, but didn't wanna complicate things more than required for me. Being able to change the network names is enough for me.

A list of blocks like network_adapters would be the right way to go here given that order between the two fields are critical to getting the networks setup correctly. I can easily see myself getting these two fields out of sync; especially if I only want to reconfigure a subset of networks. 

I would recommend creating a new BlockList similar to `network_adapters` but named differently that allows for defining the network name and mac address as a tuple to ensure the network information states together. 

For example

```hcl
addition_networks = {
 { 
   network = "new_network_for_network_adapter_1",
   mac_address = "<mac_address_for_network_adapter_1",
},
 { 
   network = "new_network_for_network_adapter_2",
   mac_address = "<mac_address_for_network_adapter_2",
},

}
```

Then internally in the driver code for the clone you could merge any settings defined under network with the context of the additional_networks maps reconfigure each of the networks as you are doing with a few modifications to iterate over a list of maps as opposed to a list of strings. 

From a backwards compatibility standpoint this would just work as it is a new field. If there was a need for network and additional_networks to be mutually exclusive (you can't define the same network in additional_networks) this could be handled by a validation the Clone Prepare method.